share/lisp/core.clp

```diff
@@ -18,19 +18,19 @@
 
 ; Do not use recursive function,
 ; since we do not support optimazation of tail call.
-(defn reduce [afn init alist]
+(defn reduce [areducefn init alist]
   (def res init)
   (def i 0)
   (def l (length alist))
   (while (< i l)
-    (set! res (afn (#:i alist) res))
+    (set! res (areducefn (#:i alist) res))
     (set! i (+ i 1)))
   res)
 
-(defn map [afn alist]
+(defn map [amapfn alist]
```

Comment:
the fact I have to rename a 'lexical' variable name seems really wrong here.  My weakening of the context looks like it does not produce the correct context for he Map function.